test/tests/supportTest.js

```diff
@@ -11,4 +11,185 @@ describe("Support Functions for Unit Testing", function() {
+				}
+			});
+			
+			let zItem = yield Zotero.Items.getAsync(data.itemWithTags.id);
+			assert.ok(zItem, 'inserted item with tags into database');
+			
+
+			let tags = data.itemWithTags.tags;
+			for (let i=0; i<tags.length; i++) {
+				let tagID = Zotero.Tags.getID(zItem.libraryID, tags[i].tag);
+				assert.ok(tagID, '"' + tags[i].tag + '" tag was inserted into the database');
+				assert.ok(zItem.hasTag(tags[i].tag), '"' + tags[i].tag + '" tag was assigned to item');
+			}
+		}));
+	});
+	describe("generateAllTypesAndFieldsData", function() {
+		it("should generate all types and fields data", function() {
+			let data = generateAllTypesAndFieldsData();
+			assert.isObject(data, 'created data object');
+			assert.isNotNull(data);
+			assert.isAbove(Object.keys(data).length, 0, 'data object is not empty');
+		});
+		it("all types and fields sample data should be up to date", function() {
+			assert.deepEqual(loadSampleData('allTypesAndFields'), generateAllTypesAndFieldsData());
+		});
+	});
+	describe("generateItemJSONData", function() {
+		it("item JSON data should be up to date", Zotero.Promise.coroutine(function* () {
+			let oldData = loadSampleData('itemJSON'),
+				newData = yield generateItemJSONData();
+			
+			assert.isObject(newData, 'created data object');
+			assert.isNotNull(newData);
+			assert.isAbove(Object.keys(newData).length, 0, 'data object is not empty');
+			
+			// Ignore data that is not stable, but make sure it is set
+			let ignoreFields = ['dateAdded', 'dateModified', 'key'];
+			for (let itemName in oldData) {
+				for (let i=0; i<ignoreFields.length; i++) {
+					let field = ignoreFields[i]
+					if (oldData[itemName][field] !== undefined) {
+						assert.isDefined(newData[itemName][field], field + ' is set');
+						delete oldData[itemName][field];
+						delete newData[itemName][field];
+					}
+				}
+			}
+
+			assert.deepEqual(oldData, newData);
+		}));
+	});
+	// describe("generateCiteProcJSExportData", function() {
+	// 	let citeURL = Zotero.Prefs.get("export.citePaperJournalArticleURL");
+	// 	before(function () {
+	// 		Zotero.Prefs.set("export.citePaperJournalArticleURL", true);
+	// 	});
+	// 	after(function() {
+	// 		Zotero.Prefs.set("export.citePaperJournalArticleURL", citeURL);
+	// 	});
+		
+	// 	it("all citeproc-js export data should be up to date", Zotero.Promise.coroutine(function* () {
+	// 		let oldData = loadSampleData('citeProcJSExport'),
+	// 			newData = yield generateCiteProcJSExportData();
+			
+	// 		assert.isObject(newData, 'created data object');
+	// 		assert.isNotNull(newData);
+	// 		assert.isAbove(Object.keys(newData).length, 0, 'citeproc-js export object is not empty');
+			
+	// 		// Ignore item ID
+	// 		for (let itemName in oldData) {
+	// 			delete oldData[itemName].id;
+	// 		}
+	// 		for (let itemName in newData) {
+	// 			delete newData[itemName].id;
+	// 		}
+			
+	// 		assert.deepEqual(oldData, newData, 'citeproc-js export data has not changed');
+	// 	}));
+	// });
```

Comment:
Still need to figure out what to do with citeproc-js, since it wants items to be retrieved synchronously. Probably we should generate the CSL JSON before calling citeproc-js.